packages/expo-sqlite/android/src/main/java/expo/modules/sqlite/SQLiteModuleNext.kt

```diff
@@ -187,16 +194,20 @@ class SQLiteModuleNext : Module() {
   }
 
   @Throws(OpenDatabaseException::class)
-  private fun pathForDatabaseName(name: String): String {
-    if (name == MEMORY_DB_NAME) {
-      return name
+  private fun ensureDatabasePathExists(databasePath: String): String {
+    if (databasePath == MEMORY_DB_NAME) {
+      return databasePath
     }
     try {
-      val directory = File("${context.filesDir}${File.separator}SQLite")
-      ensureDirExists(directory)
-      return "$directory${File.separator}$name"
-    } catch (_: IOException) {
-      throw OpenDatabaseException(name)
+      val parsedPath =
+        Uri.parse(databasePath).path ?: throw IOException("Couldn't parse Uri - $databasePath")
+      val path = File(parsedPath)
+      val parentPath =
+        path.parentFile ?: throw IOException("Parent directory is null for path '$path'.")
+      ensureDirExists(parentPath)
+      return path.canonicalPath
+    } catch (e: IOException) {
+      throw OpenDatabaseException(databasePath, e.message)
```

Comment:
Previously the original `IOException` message was lost but it could be helpful for debugging/reporting